kernel/core/maths/matfuncs.hpp

```diff
@@ -3567,6 +3567,174 @@ static Matrix matrixFilter(const MatFuncData& funcData, const MatFuncErrorInfo&
 }
 
 
+/////////////////////////////////////////////////
+/// \brief Function to perform a circular shift
+/// of the elements in a matrix by n rows.
+///
+/// \param funcData const MatFuncData&
+/// \param errorInfo const MatFuncErrorInfo&
+/// \return Matrix
+///
+/////////////////////////////////////////////////
+static Matrix circShiftRow(const MatFuncData& funcData, const MatFuncErrorInfo& errorInfo)
+{
+    // Inputs: funcData.mat1 funcData.nVal
+    std::vector<mu::value_type> matData = funcData.mat1.data();
+
+    // Circular shift downwards (negative values upwards)
+    // TODO: remove the unnecessary int casts
+    int shift = (int)funcData.nVal % (int)matData.size();
+    shift = - shift;
+    shift = (shift + (int)matData.size()) % (int)matData.size();
+
+    std::rotate(matData.begin(), matData.begin() + shift, matData.end());
+
+    Matrix _mResult = Matrix();
+    _mResult.assign(funcData.mat1.rows(), funcData.mat1.cols(), matData);
+
+    return _mResult;
+}
+
+
+/////////////////////////////////////////////////
+/// \brief Function to perform a circular shift
+/// of the elements in a matrix by n columns.
+///
+/// \param funcData const MatFuncData&
+/// \param errorInfo const MatFuncErrorInfo&
+/// \return Matrix
+///
+/////////////////////////////////////////////////
+static Matrix circShiftCol(const MatFuncData& funcData, const MatFuncErrorInfo& errorInfo)
+{
+    // Transpose the matrix (including restructuring the memory) to get a row wise representation of the data
+    Matrix _mResult = Matrix(funcData.mat1);
+    _mResult.transpose();
+    _mResult.extend(_mResult.rows(), _mResult.cols());
+
+    MatFuncData funcDataTransposed = MatFuncData(_mResult, funcData.nVal);
+
+    // Apply the circShiftRow to the transposed matrix
+    _mResult = circShiftRow(funcDataTransposed, errorInfo);
+
+    // Reverse the transposition
+    _mResult.transpose();
+    _mResult.extend(_mResult.rows(), _mResult.cols());

```

Comment:
```suggestion
    _mResult.transpose();

    // Apply the circShiftRow to the transposed matrix
    _mResult = circShiftRow(MatFuncData(std::move(_mResult), funcData.nVal), errorInfo);

    // Reverse the transposition
    _mResult.transpose();
```

This should avoid uneccessary copies as the move constructor is called twice.